contract_sale_payment_mode/models/contract_contract.py

```diff
@@ -0,0 +1,15 @@
+from odoo import api, models
+
+
+class ContractContract(models.Model):
+    _inherit = "contract.contract"
+
+    @api.depends("partner_id", "contract_type")
+    def _compute_payment_mode_id(self):
+        # Exclude contracts that already have a payment mode
+        contracts_to_compute = self.filtered(lambda c: not c.payment_mode_id)
+        if contracts_to_compute:
+            return super(
+                ContractContract, contracts_to_compute
+            )._compute_payment_mode_id()
+        return None
```

Comment:
You don't need to return None for an empty recordset. It is not a problem if super() is called on an empty recordset.
```suggestion
        return super(
            ContractContract, contracts_to_compute
        )._compute_payment_mode_id()
```

Reply:
@marielejeune done